pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -403,7 +386,7 @@ private ParsedTimestamp parseBackendTimestamp(String str) throws SQLException {
     }
 
     ParsedTimestamp ts = parseBackendTimestamp(s);
-    Calendar useCal = ts.tz != null ? ts.tz : setupCalendar(cal);
+    Calendar useCal = ts.hasOffset ? getCalendar(ts.offset) : setupCalendar(cal);
```

Comment:
Hi Dave,
what regression do you mean, because this change (at öeats this line is not anything that could break the parsing). The code change here is required to get rid of java.util.Calendar when just parsing the datestamp to ParsedTimestamp. In the previous code `ParsedTimestamp#tz` was a `Calendar` instance and not a plain timezone offset. For parsing a String date/time to an `OffsetTime` the calendar is not helpful. 

So the refactoring was needed to simple save the raw time offset in ParsedTimestamp and mve the code using the calendar to somewhere else. If there is a bug in it it would be good to see an example of the regression you are seing.

Reply:
Are you referring to this comment? https://github.com/pgjdbc/pgjdbc/pull/2476#discussion_r848790128

Reply:
From what I understand the issue in #2476 is a different one. It is about LocalDate and BCE. @vlsi mentioned that we want to possibly fix it. I don't think this has anything to do with this PR, it may be caused by the previous one (#2464)?

Reply:
I think we should add a test that fails and open a separate PR to fix a possible year shift for BC LocalDate (caused by a different defintiion of Year 0 in Postgresql and Java).

Reply:
When you give me an hint about the exact problem, I can figure out if we maybe should fix this like proposed here: https://github.com/pgjdbc/pgjdbc/pull/2467/files/918926e6e92789381058dd8b1ea2946f2329dc70#r833415824